osu.Game.Modes.Osu/Objects/SliderCurve.cs

```diff
@@ -30,7 +30,7 @@ private List<Vector2> calculateSubpath(List<Vector2> subControlPoints)
                     return subControlPoints;
                 case CurveTypes.PerfectCurve:
                     // If we have a different amount than 3 control points, use bezier for perfect curves.
-                    if (ControlPoints.Count != 3)

```

Comment:
We only want the arc approximator if there is a _single_ segment with 3 control points.